ocs_ci/ocs/resources/mcg.py

```diff
@@ -168,3 +193,81 @@ def cli_verify_bucket_exists(self, bucketname):
 
         """
         return bucketname in self.cli_list_all_bucket_names()
+
+    def send_rpc_query(self, api, method, params):
+        """
+        Templates and sends an RPC query to the MCG mgmt endpoint
+
+        Args:
+            api: The name of the API to use
+            method: The method to use inside the API
+            params: A dictionary containing the command payload
+
+        Returns:
+            The server's response
+
+        """
+        payload = {
+            'api': api,
+            'method': method,
+            'params': params,
+            'auth_token': self.noobaa_token
+        }
+        return requests.post(url=self.mgmt_endpoint, data=json.dumps(payload), verify=False)
+
+    def check_data_reduction(self, bucketname):
+        """
+        Checks whether the data reduction on the MCG server works properly
+        Args:
+            bucketname: An example bucket name that contains compressed/deduped data
+
+        Returns:
+            bool: True if the data reduction mechanics work, False otherwise
+
+        """
+
+        def _retrieve_reduction_data():
+            payload = {
+                "api": "bucket_api",
+                "method": "read_bucket",
+                "params": {"name": bucketname},
+                "auth_token": self.noobaa_token
+            }
+            request_str = json.dumps(payload)
+            resp = requests.post(url=self.mgmt_endpoint, data=request_str, verify=False)
+            bucket_data = resp.json().get('reply').get('data').get('size')
+
+            payload = {
+                "api": "bucket_api",
+                "method": "read_bucket",
+                "params": {"name": bucketname},
+                "auth_token": self.noobaa_token
+            }
+            request_str = json.dumps(payload)
+            resp = requests.post(url=self.mgmt_endpoint, data=request_str, verify=False)
+            bucket_data_reduced = resp.json().get('reply').get('data').get('size_reduced')
+
+            logger.info(
+                'Overall bytes stored: ' + str(bucket_data) + '. Amount reduced: ' + str(bucket_data_reduced)
+            )
+
+            return bucket_data, bucket_data_reduced
```

Comment:
this function should return bool

Reply:
Woah, that line should've been deleted. Fixed. Thank you!

Reply:
Please note this is the inside function, `_retrieve_data_reduction`. It retrieves the data reduction data - `check_data_reduction` returns bool

Reply:
ack